guides/cheatsheets/git.md

```diff
@@ -0,0 +1,305 @@
+    git config --local user.email “[valid-email]”
+
+## **SETUP & INIT**
+
+Below are commands for initializing and cloning repositories.
+
+_Initialise an existing directory as a Git repository_.
+
+    git init
+
+_Retrieve an entire repository from a hosted location via URL_.
+
+    git clone [url]
+
+---
+
+## **STAGE & SNAPSHOT**
+
+---
+
+Working with snapshots and the Git staging area.
+
+_Show modified files in working directory, staged for your next commit_.
+
+    git status
+
+_Add a file as it looks now to your next commit (stage)_.
+
+    git add [file]
+
+_Add everything your next commit_.
+
+    git add .
+
+_Commit your staged content as a new commit snapshot_.
+
+    git commit -m “[descriptive message]”
+
+_Upload any changes to your local repository to your remote repository_.
+
+    git push
+
+_Upload any changes to your local repository to your remote repository,
+rewriting the git history (index)_.
+
+NOTE: It's normally discouraged except for certain use cases like squashing
+commits.
+
+    git push -f
+
+_If you find yourself in a situation where you are required to do a force push,
+you can push multiple branches and only force one like so: (+ denotes that the
+following branch will be force pushed (issue/1))_.
+
+    git push origin +issue/1 issue/2
+
+_Unstage a file while retaining the changes in working directory_.
+
+    git reset [file]
+
+_Diff of what is changed but not staged_.
+
+    git diff
+
+_Diff of what is staged but not yet commited_.
+
+    git diff --staged
+
+---
+
+## **BRANCH & MERGE**
+
+---
+
+Isolating work in branches, changing context, and integrating changes.
+
+_List your branches. a\*will appear next to the currently active branch\._
+
+    git branch
+
+_Create a new branch_.
+
+    git branch [branch-name]
+
+_Switch to another branch and check it out into your working directory_.
+
+    git checkout
+
+_Switch to the specified branch. Creates the branch in your local repository if
+it doesn't already exist_.
+
+    git checkout -b [branch_name]
+
+_Merge the specified branch’s history into the current one_.
+
+    git merge [branch]
+
+_Show all commits in the current branch’s history_.
+
+    git log
+
+---
+
+## **INSPECT & COMPARE**
+
+---
+
+Examining logs, diffs and object information.
+
+_Show the commit history for the currently active branch_.
+
+    git log
+
+_Show the commits on branchA that are not on branchB_.
+
+    git log branchB..branchA
+
+_Show the commits that changed file, even across renames_.
+
+    git log --follow [file]
+
+_Get the author of a commit_.
+
+    git blame [commit_hash]
+
+_Show the diff of what is in branchA that is not in branchB_.
+
+    git diff branchB...branchA
+
+_Show any object in Git in human-readable format_.
+
+    git show [SHA]
+
+---
+
+## **TRACKING PATH CHANGES**
+
+---
+
+Versioning file removes and path changes.
+
+_Delete the file from project and stage the removal for commit_.
+
+    git rm [file]
```

Comment:
`git rm --cached` will remove it from Git's control but leave the file in place: occasionally useful.